### PR TITLE
[ST] feature gate KRaftMode test enabled 

### PR DESCRIPTION
**Pull request by henryZrncik**
### Type of change

- Refactoring

### Description

`testKRaftMode` enabled,  in addition to `kraft` feature gate also  `KafkaNodePools` are used in this case. 

fixes  https://github.com/strimzi/strimzi-kafka-operator/issues/8827

### CODE REVIEW

**Comment by henryZrncik** [comment]
@strimzi-ci run tests --testcase=FeatureGatesIsolatedST

File: systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesIsolatedST.java [hunk 1]
```diff
@@ -117,7 +123,21 @@ public void testKRaftMode(ExtensionContext extensionContext) {
             .endSpec()
             .build();
         kafka.getSpec().getEntityOperator().setTopicOperator(null); // The builder cannot disable the EO. It has to be done this way.
-        resourceManager.createResource(extensionContext, kafka);
+
+        KafkaNodePool kafkaNodePool = KafkaNodePoolResource.convertKafkaResourceToKafkaNodePool(kafka);
+        kafkaNodePool = new KafkaNodePoolBuilder(kafkaNodePool)
+            .editOrNewMetadata()
+                .addToLabels(Labels.STRIMZI_CLUSTER_LABEL, testStorage.getClusterName())
+            .endMetadata()
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesIsolatedST.java:131]: This is already added in `convertKafkaResourceToKafkaNodePool`

**Review by im-konge — APPROVED** [review]
LGTM, just one nit.

**Comment by im-konge** [comment]
There is one failure -`ConnectBuilderIsolatedST.testUpdateConnectWithAnotherPlugin` - which is not related to this PR. I guess we are good to go with this.

**Pull request merged**